### ISSUE TITLE
Error: _gui.free_disk_space: Function: Overwrite free disk space

### ISSUE THREAD

**Issue by ediowar** [bleachbit/bleachbit#1043]
git revision https://github.com/bleachbit/bleachbit/commit/17e0c0e512ba33b6e579e066559a883d5ee88d08 Ubuntu 16.04

Wipe Free Space
Disks info: Thumb Drive 8,0 GB (8.040.480.768 bytes) FAT
Two files are written but not deleted.:
File1 size 4,0 GB (4 294 967 295 bytes)
File2 size 3,5 GB (3 729 817 600 bytes)

```
dzmitry@dzmitry-Satellite-C50-A-L6K:~/Downloads/bleachbit-master$ sudo python3 bleachbit.py
clean_operation('_gui'), options = '[]'
Wiping path: /media/dzmitry/6A17-921B
Creating new, temporary file for wiping free space.
Error #27 when flushing the file buffer.
Creating new, temporary file for wiping free space.
Wrote 2 files and 8,024,788,992 bytes in 892 seconds at 9.00 MB/s
0 bytes and 0 inodes available to non-super-user
0 bytes and 0 inodes available to super-user
Error: _gui.free_disk_space: Function: Overwrite free disk space /media/dzmitry/6A17-921B
Traceback (most recent call last):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/Worker.py", line 87, in execute
    for ret in cmd.execute(self.really_delete):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/Command.py", line 145, in execute
    for func_ret in self.func():
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/Cleaner.py", line 743, in wipe_path_func
    for ret in FileUtilities.wipe_path(path, idle=True):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/FileUtilities.py", line 984, in wipe_path
    truncate_f(f)
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/FileUtilities.py", line 680, in truncate_f
    f.truncate(0)
  File "/usr/lib/python3.5/tempfile.py", line 622, in func_wrapper
    return func(*args, **kwargs)
OSError: [Errno 27] File too large
Exception ignored in: <bound method _TemporaryFileCloser.__del__ of <tempfile._TemporaryFileCloser object at 0x7fdac08e44a8>>
Traceback (most recent call last):
  File "/usr/lib/python3.5/tempfile.py", line 589, in __del__
    self.close()
  File "/usr/lib/python3.5/tempfile.py", line 582, in close
    self.file.close()
OSError: [Errno 28] No space left on device
Exception ignored in: <bound method _TemporaryFileCloser.__del__ of <tempfile._TemporaryFileCloser object at 0x7fdab9fb7ac8>>
Traceback (most recent call last):
  File "/usr/lib/python3.5/tempfile.py", line 589, in __del__
    self.close()
  File "/usr/lib/python3.5/tempfile.py", line 582, in close
    self.file.close()
OSError: [Errno 27] File too large
elapsed time: 892 seconds
```
```
sys.version = 3.5.2 (default, Oct  7 2020, 17:19:02) 
```

**Comment by Marvin0401** [comment]
Hi Andrew, Let me fix this issue. Are you okay?

**Comment by ediowar** [comment]
@Marvin0401 https://github.com/Marvin0401/bleachbit/commit/6c60e606dd2bc233f4e780d802a2628b1978400f
```
dzmitry@dzmitry-Satellite-C50-A-L6K:~/Downloads/bleachbit-master$ sudo python3 bleachbit.py
![Screenshot from 2020-12-09 01-47-40](https://user-images.githubusercontent.com/44164186/101550638-b5b56480-39c0-11eb-956f-838a9bd6b187.png)

clean_operation('_gui'), options = '[]'
Wiping path: /media/dzmitry/6C91-432C
Creating new, temporary file for wiping free space.
Error #27 when flushing the file buffer.
Creating new, temporary file for wiping free space.
Wrote 2 files and 8,023,736,320 bytes in 1,181 seconds at 6.79 MB/s
0 bytes and 0 inodes available to non-super-user
0 bytes and 0 inodes available to super-user
Error: _gui.free_disk_space: Function: Overwrite free disk space /media/dzmitry/6C91-432C
Traceback (most recent call last):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/Worker.py", line 87, in execute
    for ret in cmd.execute(self.really_delete):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/Command.py", line 145, in execute
    for func_ret in self.func():
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/Cleaner.py", line 743, in wipe_path_func
    for ret in FileUtilities.wipe_path(path, idle=True):
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/FileUtilities.py", line 988, in wipe_path
    truncate_f(f)
  File "/home/dzmitry/Downloads/bleachbit-master/bleachbit/FileUtilities.py", line 680, in truncate_f
    f.truncate(0)
  File "/usr/lib/python3.8/tempfile.py", line 613, in func_wrapper
    return func(*args, **kwargs)
OSError: [Errno 27] File too large
Exception ignored in: <function _TemporaryFileCloser.__del__ at 0x7f92313de310>
Traceback (most recent call last):
  File "/usr/lib/python3.8/tempfile.py", line 580, in __del__
    self.close()
  File "/usr/lib/python3.8/tempfile.py", line 573, in close
    self.file.close()
OSError: [Errno 28] No space left on device
Exception ignored in: <function _TemporaryFileCloser.__del__ at 0x7f92313de310>
Traceback (most recent call last):
  File "/usr/lib/python3.8/tempfile.py", line 580, in __del__
    self.close()
  File "/usr/lib/python3.8/tempfile.py", line 573, in close
    self.file.close()
OSError: [Errno 27] File too large
elapsed time: 1182 seconds

```
![Peek 2020-12-09 01-22](https://user-images.githubusercontent.com/44164186/101549305-7d148b80-39be-11eb-9c91-ddc9773e61ce.gif)
![Screenshot from 2020-12-09 01-47-40](https://user-images.githubusercontent.com/44164186/101550679-ce257f00-39c0-11eb-9539-69d53e69c454.png)
When exiting Bleachbit, the files are deleted.

**Comment by abitrolly** [comment]
https://docs.python.org/2/library/stdtypes.html#file.truncate says that `The current file position is not changed.`. I would try replacing this line.

https://github.com/bleachbit/bleachbit/blob/master/bleachbit/FileUtilities.py#L680
```
- f.truncate(0)
+ f.seek(0)
+ f.truncate()
```

On Linux it is also also possible to run `strace bleachbit` to what is really going on.

**Comment by Marvin0401** [comment]
Hi @ediowar , 
I've fixed this issue. Could you please check it with this repository, Marvin0401@ae4375a?

**Comment by Marvin0401** [comment]
In the case of FAT32 system in Ubuntu, only files smaller than 4GB can be managed. If the file size is more than 4GB, an EFBIG error occurs.

**Comment by ediowar** [comment]
@Marvin0401 ,
[Terminal messages.txt](https://github.com/bleachbit/bleachbit/files/5824626/Terminal.messages.txt)

**Comment by az0** [comment]
> In the case of FAT32 system in Ubuntu, only files smaller than 4GB can be managed. If the file size is more than 4GB, an EFBIG error occurs.

The expected behavior is that BleachBit quietly creates another 4GiB file without deleting the first file. This continues until the partition is full.

**Comment by Marvin0401** [comment]
> > In the case of FAT32 system in Ubuntu, only files smaller than 4GB can be managed. If the file size is more than 4GB, an EFBIG error occurs.
> 
> The expected behavior is that BleachBit quietly creates another 4GiB file without deleting the first file. This continues until the partition is full.

Yes, it is working now as you think. I meant a file size is smaller than 4GB

**Comment by Marvin0401** [comment]
@ediowar ,
Could you please tell me the size of two files created at run time?
I hope to know the file system type of your disk. There is a get_filesystem_type function in my repository. Please let me know the return value of this function at your runtime. Thank you.

**Comment by ediowar** [comment]
@Marvin0401 ,
Two files are written but not deleted.:
File1 size 4.294.967.295 bytes
File2 size 3.728.764.928 bytes
After installing the psutil package, there is no error, vfat.
File1 size 4 294 901 760 bytes
Wrote 2 files and 8,023,736,320 bytes

**Comment by ediowar** [comment]
@Marvin0401 ,
Size of disk 8.040.480.768 bytes
Two files are written but not deleted.:
File1 size 4.294.967.295 bytes
File2 size 3.728.764.928 bytes
After installing the psutil package, there is no error.
File System:vfat
File1 size 4 294 901 760 bytes
Wrote 2 files and 8,023,736,320 bytes

**Comment by Marvin0401** [comment]
Ok. Thank you. I've used psutil package to get the file system type. Thank you again.

**Comment by ediowar** [comment]
I tested the fix for issue #1043 issue running from latest source code git revision https://github.com/bleachbit/bleachbit/commit/86a71599b7b00793f2052b8c6facca7ebe14a93b (with this commit) on Ubuntu 20.04 with an 8GB FAT32 flash drive. It correctly created two temporary files. Here's the log.
```
clean_operation('_gui'), options = '[]'
Wiping free disk space erases remnants of files that were deleted without shredding. It does not free up space.Wiping path: /media/dzmitry/22DC-13D2
Creating new, temporary file for wiping free space.
Creating new, temporary file for wiping free space.
Wrote 2 files and 8,022,654,976 bytes in 2,096 seconds at 3.83 MB/s
0 bytes and 0 inodes available to non-super-user
0 bytes and 0 inodes available to super-user
Overwrite free disk space /media/dzmitry/22DC-13D2 0 

Disk space recovered: 0
Files deleted: 0
Special operations: 1
elapsed time: 2105 seconds
```

**Comment by Marvin0401** [comment]
Thank you for your testing.
